### PR TITLE
PyTimeDelta class for py.js

### PR DESCRIPTION
**Pull request by Polymorphe57**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/core/py_js/py_date.js [hunk 1]
```diff
@@ -500,3 +500,153 @@ export class PyRelativeDelta {
...
     }
 }
+
+const TIME_DELTA_KEYS = "weeks days hours minutes seconds milliseconds microseconds".split(" ");
+
+/**
+ * Passes the fractional and integer parts of x to the callback, returns
+ * the callback's result
+ */
+function modf(x, fn) {
```
Comment (ged-odoo) [inline addons/web/static/src/core/py_js/py_date.js:510]: why not use a synchronous api? it's just a pure function
```js
function modf(x) {
  const mod = x % 1;
  return [mod < 0 ? mod + 1 : mod, Math.floor(x)];
```

// can be used like this:
let [value, floor] = modf(x);

Comment (ged-odoo) [reply addons/web/static/src/core/py_js/py_date.js:510]: by the way, maybe we should do that with divmod as well. it makes it so complicated to have a callback function

File: addons/web/static/src/core/py_js/py_interpreter.js [hunk 2]
```diff
@@ -99,6 +113,10 @@ function isEqual(left, right) {
         }
         return false;
     }
+    if (left instanceof Object && left.isEqual) {
```
Comment (Polymorphe57) [inline addons/web/static/src/core/py_js/py_interpreter.js:116]: not sure about this

File: addons/web/static/src/core/py_js/py_date.js (hunk 1 above)
Comment (Polymorphe57) [reply addons/web/static/src/core/py_js/py_date.js]: not done in the commit. Maybe after that?

Comment (Polymorphe57) [reply addons/web/static/src/core/py_js/py_date.js]: > why not use a synchronous api? it's just a pure function
> 
> ```js
> function modf(x) {
>   const mod = x % 1;
>   return [mod < 0 ? mod + 1 : mod, Math.floor(x)];
> }
> 
> // can be used like this:
> let [value, floor] = modf(x);
> ```

donde

File: addons/web/static/src/core/py_js/py_date.js [hunk 3]
```diff
@@ -325,14 +374,60 @@ export class PyDateTime {
...
+        // does not seem to closely follow python implementation.
+        return new PyDateTime(s.year, s.month, s.day, s.hour, s.minute, s.second, s.microsecond);
+    }
+
+    /**
+     * @param {any} other
+     * @returns {boolean}
+     */
+    isEqual(other) {
+        if (!(other instanceof PyDateTime)) {
```
Comment (ged-odoo) [inline addons/web/static/src/core/py_js/py_date.js:419]: no need for check

File: addons/web/static/src/core/py_js/py_date.js [hunk 4]
```diff
@@ -500,3 +600,165 @@ export class PyRelativeDelta {
...
+        const us = (this.days * 24 * 3600 + this.seconds) * 1e6 + this.microseconds;
+        return PyTimeDelta.create({ microseconds: Math.floor(us / n) });
+    }
+
+    /**
+     * @param {any} other
+     * @returns {boolean}
+     */
+    isEqual(other) {
+        if (!(other instanceof PyTimeDelta)) {
```
Comment (ged-odoo) [inline addons/web/static/src/core/py_js/py_date.js:706]: no need for check

**Comment by ged-odoo** [comment]
robodoo r+

**Comment by ged-odoo** [comment]
robodoo r+

**Pull request closed without merging**